### ISSUE TITLE
FFT backend for Filter2d

### ISSUE THREAD

**Issue by oliland** [kornia/kornia#1731]
## 🚀 Feature
An FFT backend for `kornia.filters.filter_2d`.

## Motivation

`kornia.filters.filter_2d` is very slow for large kernels.

## Pitch

Convolution in the spatial domain is equivalent to multiplication in the Fourier domain. For large kernels, transforming both the kernel and the image to the fourier domain, multiplying, and inverting the FFT should yield a much faster `filter2d` implementation for large kernels.

## Alternatives

-

## Additional context

-

**Comment by edgarriba** [comment]
@oliland that would be an awesome contribution. I guess we are targeting something like this:
https://stackoverflow.com/questions/60561933/verify-convolution-theorem-using-pytorch

A benchmark should be included. I also noticed some slow downs when using large kenels in gaussian blur and similar

**Comment by edgarriba** [comment]
on the other hand, not sure if pytorch itself exposes the algorithm to use for the conv2d /cc @ducha-aiki @t-vi

**Comment by ducha-aiki** [comment]
I think, that if you run it with `torch.backends.cudnn.benchmark=True`, it would allow PyTorch to pickup the right alg. I am not sure if conv2d allows to select an algorithm, but I know for sure there is FFT implementation among cudnn algs.

**Comment by edgarriba** [comment]
@ducha-aiki this will be valid only for cuda/nvidia hardware

**Comment by ducha-aiki** [comment]
Yes, but for others it would be even more complicated, we would have to have many ifs per backend per kernels size then
I think that CPU conv is done via im2col https://discuss.pytorch.org/t/how-was-conv2d-implemented-in-pytorch/35223

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!

**Comment by manas1245agrawal** [comment]
Hello can I reopen the ticket
